homeassistant/components/vulcan/config_flow.py

```diff
@@ -0,0 +1,342 @@
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PIN, CONF_REGION, CONF_SCAN_INTERVAL, CONF_TOKEN
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+
+from . import DOMAIN
+from .const import DEFAULT_SCAN_INTERVAL
+from .register import register
+
+_LOGGER = logging.getLogger(__name__)
+
+LOGIN_SCHEMA = {
+    vol.Required(CONF_TOKEN): str,
+    vol.Required(CONF_REGION): str,
+    vol.Required(CONF_PIN): str,
+}
+
+
+class VulcanFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Uonet+ Vulcan config flow."""
+
+    VERSION = 1
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return VulcanOptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle config flow."""
+        if self._async_current_entries():
+            return await self.async_step_add_next_config_entry()
+
+        return await self.async_step_auth()
+
+    async def async_step_auth(self, user_input=None, errors=None):
+        """Authorize integration."""
+
+        if user_input is not None:
+            try:
+                credentials = await register(
+                    self.hass,
+                    user_input[CONF_TOKEN],
+                    user_input[CONF_REGION],
+                    user_input[CONF_PIN],
+                )
+            except VulcanAPIException as err:
+                if str(err) == "Invalid token!" or str(err) == "Invalid token.":
+                    errors = {"base": "invalid_token"}
+                elif str(err) == "Expired token.":
+                    errors = {"base": "expired_token"}
+                elif str(err) == "Invalid PIN.":
+                    errors = {"base": "invalid_pin"}
+                else:
+                    errors = {"base": "unknown"}
+                    _LOGGER.error(err)
+            except RuntimeError as err:
+                if str(err) == "Internal Server Error (ArgumentException)":
+                    errors = {"base": "invalid_symbol"}
+                else:
+                    errors = {"base": "unknown"}
+                    _LOGGER.error(err)
+            except ClientConnectionError as err:
+                errors = {"base": "cannot_connect"}
+                _LOGGER.error("Connection error: %s", err)
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors = {"base": "unknown"}
+            if not errors:
+                account = credentials["account"]
+                keystore = credentials["keystore"]
+                client = Vulcan(keystore, account)
+                students = await client.get_students()
+                await client.close()
+
+                if len(students) > 1:
+                    # pylint:disable=attribute-defined-outside-init
```

Comment:
Don't disable this. Please fix the issue instead and define all attributes properly in init method.